io/js/src/main/scala/fs2/io/ioplatform.scala

```diff
@@ -41,12 +41,14 @@ import scala.scalajs.js.|
 
 private[fs2] trait ioplatform {
 
-  def fromReadable[F[_]](readable: F[Readable])(implicit F: Async[F]): Stream[F, Byte] =
+  def readReadable[F[_]](readable: F[Readable], destroyIfNotEnded: Boolean = true)(implicit
+      F: Async[F]
+  ): Stream[F, Byte] =
     Stream
       .resource(for {
         readable <- Resource.makeCase(readable.map(_.asInstanceOf[streamMod.Readable])) {
           case (readable, Resource.ExitCase.Succeeded) =>
-            if (!readable.readableEnded)
+            if (!readable.readableEnded & destroyIfNotEnded)
               F.delay(readable.destroy())
             else
               F.unit
```

Comment:
Still not thrilled with this but I think it is the best we can do. The underlying issue is that Node.js has no way to close/"end" a `Readable` besides the nuclear option `destroy`. This same thing came up in https://github.com/typelevel/fs2/pull/2453#discussion_r666153173 when we couldn't implement the `endOfInput` method for sockets.